components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3501,14 +3535,106 @@ func TestListRules(t *testing.T) {
 			rule2, err := storage.NewRule("new-id-2", projID2, "name2", ruleType,
 				[]storage.Condition{condition4})
 			require.NoError(t, err)
-			insertAppliedRule(t, db, rule2)
+			insertAppliedRule(t, db, &rule2)
 			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule1.ID))
 			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule2.ID))
 			assertCount(t, 4, db.QueryRow(`SELECT count(*) FROM iam_rule_conditions`))
 
 			resp, err := store.ListRules(ctx)
 			assert.NoError(t, err)
-			assert.ElementsMatch(t, []*storage.Rule{&rule2}, resp)
+			require.NotZero(t, len(resp))
+			assert.Equal(t, &rule2, resp[0])
+		},
+	}
+
+	for name, test := range cases {
+		t.Run(name, test)
+		db.Flush(t)
+	}
+}
+func TestListStagedAndAppliedRules(t *testing.T) {
```

Comment:
The amount of copy-paste involved in creating these tests is deeply unsatisfying :grumpycat: